lib/js/stanford_publication.behavior.js

```diff
@@ -13,20 +13,48 @@ export default {
   attach: function (context, settings) {
     (function ($) {
 
-      $('#system-menu-blockstanford-publication-topics-menu')
-        .wrap('<button class="publication-topics__collapsable-menu"></button>');
-
-      if ($(window).width() <= 991) {
-        $('.publication-topics__collapsable-menu', context).click(function () {
-          $(this).toggleClass('show');
-          if ($(this).siblings('.menu').css('display') !== 'none') {
-            $(this).attr('aria-expanded', 'true');
-          }
-          else {
-            $(this).attr('aria-expanded', 'false');
-          }
-        });
+      // only open and close filter menu when on responsive size.
+      const mediaQuery = window.matchMedia('(max-width: 991px)')
+
+      function handleTabletChange(e) {
+        // Check if the media query is true
+        if (e.matches) {
+
+          // Remove the h2 and replace with button.
+          $("#system-menu-blockstanford-publication-topics-menu")
+            .contents()
+            .unwrap()
+            .wrap('<button class="publication-topics__collapsable-menu"></button>');
+
+          $('.publication-topics__collapsable-menu', context).click(function () {
+            $(this).toggleClass('show');
+            if ($(this).siblings('.menu').css('display') !== 'none') {
+              $(this).attr('aria-expanded', 'true');
+            }
+            else {
+              $(this).attr('aria-expanded', 'false');
+            }
+          });
+
+        }
+        else {
+          // Add the class we need for the desktop view.
+          $('#system-menu-blockstanford-publication-topics-menu').addClass('publication-topics__collapsable-menu');
+        }
       }
+
+// Register event listener
+      window.addEventListener('resize', mediaQuery);
```

Comment:
@pookmish It doesn't quite work. Do you see the problem here?

Reply:
you don't register the listener that way.
`landscapeMQ.addListener(handleTabletChange);`

see the [example in the conclusion](https://css-tricks.com/working-with-javascript-media-queries/#conclusion) for that.

Reply:
I fixed that up. I thought the addListener was deprecated. It doesn't really do what I'd like. When the window is resized, I'd like that to run again. But it doesn't.